hack/jenkins/test-flake-chart/report_flakes/report_flake.go

```diff
@@ -0,0 +1,243 @@
+					env:       env,
+					testName:  test,
+				})
+		}
+
+		sort.Slice(failedTestList, func(i, j int) bool {
+			return failedTestList[i].flakeRate < failedTestList[j].flakeRate
+		})
+		envFailedTestList[env] = failedTestList
+	}
+	// we convert the result into a 2d string slice representing a markdown table,
+	// whose each line represents a line of the table
+	table := [][]string{
+		// title of the table
+		{"Environment", "Test Name", "Flake Rate"},
+	}
+	// if an env has too much failures we will just skip it and print a message in the end
+	tooMuchFailure := []string{}
+	for env, list := range envFailedTestList {
+		if len(list) > MAX_ITEM_ENV {
+			tooMuchFailure = append(tooMuchFailure, env)
+			continue
+		}
+		for _, item := range list {
+			flakeRateString := fmt.Sprintf("%.2f%% %s", item.flakeRate, testFlakeChartMDLink(env, item.testName))
+			if item.flakeRate < 0 {
+				flakeRateString = "Unknown"
+			}
+			table = append(table, []string{
+				envChartMDLink(env, len(list)),
+				item.testName + gopoghMDLink(pr, rootJob, env, item.testName),
+				flakeRateString,
+			})
+		}
+	}
+
+	builder := strings.Builder{}
+	builder.WriteString(
+		fmt.Sprintf("Here are the number of top %d failed tests in each environments with lowest flake rate.\n\n", MAX_ITEM_ENV))
+	builder.WriteString(generateMarkdownTable(table))
+	if len(tooMuchFailure) > 0 {
+
+		builder.WriteString("\n\n Besides the following environments have too much failed tests:")
+		for _, env := range tooMuchFailure {
+			builder.WriteString(fmt.Sprintf("\n\n - %s: %d failed", env, len(envFailedTestList[env])))
+		}
+	}
+	builder.WriteString("\n\nTo see the flake rates of all tests by environment, click [here](https://minikube.sigs.k8s.io/docs/contrib/test_flakes/).")
+	return builder.String()
+}
+func envChartMDLink(env string, failedTestNumber int) string {
+	return fmt.Sprintf("[%s (%d failed)](https://gopogh-server-tts3vkcpgq-uc.a.run.app/?env=%s)", env, failedTestNumber, env)
+}
+
+func testFlakeChartMDLink(env string, testName string) string {
+	return fmt.Sprintf("[(chart)](https://gopogh-server-tts3vkcpgq-uc.a.run.app/?env=%s&test=%s)", env, testName)
+}
+
+func gopoghMDLink(pr, rootJob, env, testName string) string {
+	return fmt.Sprintf("[(gopogh)](https://storage.googleapis.com/minikube-builds/logs/%s/%s/%s.html#%s)", pr, rootJob, env, testName)
+}
+
+// generateMarkdownTable convert 2d string slice into markdown table. The first string slice is the header of the table
+func generateMarkdownTable(table [][]string) string {
+	builder := strings.Builder{}
+	for i, group := range table {
+		builder.WriteString("|")
+		for j := 0; j < len(group); j++ {
+			builder.WriteString(group[j])
+			builder.WriteString("|")
+		}
+		builder.WriteString("\n")
+
+		if i == 0 {
+			// generate the hyphens seperator
+			builder.WriteString("|")
+			for j := 0; j < len(group); j++ {
+				builder.WriteString(" ---- |")
+			}
+			builder.WriteString("\n")
+		}
```

Comment:
```suggestion
		if i != 0 {
			continue
		}
		// generate the hyphens seperator
		builder.WriteString("|")
		for j := 0; j < len(group); j++ {
			builder.WriteString(" ---- |")
		}
		builder.WriteString("\n")
```

Invert `if` to prevent nesting